### PR TITLE
Refactor ipfs get to use CoreAPI

### PR DESCRIPTION
**Pull request by magik6k**
Depends on:
* [x] https://github.com/ipfs/go-unixfs/pull/59
* [x] https://github.com/ipfs/go-ipfs-files/pull/4

### CODE REVIEW

**Comment by Stebalien** [comment]
> why put tar-writer in go-ipfs-files

Because `coreapi.Unixfs().Get()` now returns a `go-ipfs-files.Node` object. Therefore, the `TarWriter` needs to convert this (instead of a dag) into a tarfile.

However: https://github.com/ipfs/go-unixfs/pull/59#issuecomment-458731609

> Am I correct in understanding that neither of these two changes are strickly essential to delivering the refactor to make get use CoreAPI?

We need *some* way to go from a `go-ipfs-files.Node` to a tar because `coreapi.Unixfs().Get()` returns a `go-ipfs-files.Node` instead of a raw IPLD DAG. We could just keep `DagArchive` as it is as a way to go directly from a DAG to a tar file.

**Comment by magik6k** [comment]
> We could just keep DagArchive as it is as a way to go directly from a DAG to a tar file.

As a note - I decided to drop it as the combo of `NewUnixfsFile`+`files.TarWriter` do the same thing with the only difference being the `files` interface in between (there shouldn't be any meaningful performance impact). Plus after switching `ipfs get` to the new thing, it would basically become dead code.

File: core/commands/get.go [hunk 1]
```diff
@@ -66,33 +66,30 @@ may also specify the level of compression by specifying '-l=<1-9>'.
...
-			res.SetLength(size)
-		case *dag.RawNode:
-			res.SetLength(uint64(len(dn.RawData())))
-		default:
+		size, err := file.Size()
+		if err != nil {
 			return err
 		}
 
+		res.SetLength(uint64(size))
```
Comment (Stebalien) [inline core/commands/get.go]: This is actually incorrect but it's an existing bug... https://github.com/ipfs/go-ipfs/issues/5690

Comment (magik6k) [reply core/commands/get.go]: The alternative is not setting this at all, but it will break progress bar... (We could pass this in a separate header)

**Comment by Stebalien** [comment]
Error is: "Error: cannot write to readonly DAGService"

**Comment by Stebalien** [comment]
The issue is: https://github.com/ipfs/go-unixfs/blob/0eb1ef8f33371b1b9f74f74655b584144919ce91/file/unixfile.go#L121. This is causing the hamt to serialize and save itself.

**Comment by magik6k** [comment]
Ah, right, it breaks the progress bar ._.

**Comment by magik6k** [comment]
See https://github.com/ipfs/go-unixfs/pull/61

**Pull request merged**